private/serialization.nim

```diff
@@ -99,18 +127,23 @@ proc representObject*(value: int, tagStyle: TagStyle,
 {.push overflowChecks: on.}
 proc parseBiggestUInt(s: string): uint64 =
   result = 0
-  for c in s:
-    if c in {'0'..'9'}: result *= 10.uint64 + (uint64(c) - uint64('0'))
-    elif c == '_': discard
-    else: raise newException(ValueError, "Invalid char in uint: " & c)
+  if s[0] == '0' and (s[1] == 'x' or s[1] == 'X'):
+    result = parseHex[uint64](s)
+  elif s[0] == '0' and (s[1] == 'o' or s[1] == 'O'):
+    result = parseOctal[uint64](s)
+  else:
+    for c in s:
+      if c in {'0'..'9'}: result = result * 10.uint64 + (uint64(c) - uint64('0'))
+      elif c == '_': discard
+      else: raise newException(ValueError, "Invalid char in uint: " & c)
 {.pop.}
 
 proc constructObject*[T: uint8|uint16|uint32|uint64](
     s: var YamlStream, c: ConstructionContext, result: var T)
     {.raises: [YamlConstructionError, YamlStreamError].} =
   ## construct an unsigned integer value from a YAML scalar
   constructScalarItem(s, item, T):
-    result = T(parseBiggestUInt(item.scalarContent))
+    result = T(yaml.parseBiggestUInt(item.scalarContent))
```

Comment:
Without yields `private/serialization.nim(140, 32) Error: ambiguous call; both yaml.parseBiggestUInt(s: string) and strutils.parseBiggestUInt(s: string) match for: (string)` against nim 74b8216e83d15dacdf82d8a2596b95900af07c48.

With 0.14.2 compilation fails due to `private/lex.nim(802, 35) Error: type mismatch: got (proc (lex: YamlLexer): bool{.gcsafe, locks: 0.}) but expected 'LexerState = proc (lex: YamlLexer): bool{.gcsafe, locks: 0.}'`.